addons/addon-base-raas-ui/packages/base-raas-ui/src/models/aws-accounts/AwsAccountsStore.js

```diff
@@ -26,16 +32,18 @@ const filterNames = {
   ALL: 'All',
   CURRENT: 'Up-to-Date',
   UPDATEME: 'Needs Update',
-  ONBOARDME: 'Needs Onboard',
+  NEW: 'Needs Onboarding',
+  ERRORED: 'Errored',
 };
 
 // A map, with the key being the filter name and the value being the function that will be used to filter the workspace
 // cfnStackName is an empty string if the account hasn't been onboarded yet
 const filters = {
   [filterNames.ALL]: () => true,
-  [filterNames.CURRENT]: account => account.needsPermissionUpdate === false && account.cfnStackName !== '',
-  [filterNames.UPDATEME]: account => account.needsPermissionUpdate === true && account.cfnStackName !== '',
-  [filterNames.ONBOARDME]: account => account.cfnStackName === '',
+  [filterNames.CURRENT]: account => account.cfnStackName !== '' && account.permissionStatus === 'CURRENT',
+  [filterNames.UPDATEME]: account => account.cfnStackName !== '' && account.permissionStatus === 'NEEDSUPDATE',
+  [filterNames.NEW]: account => account.permissionStatus === 'NEEDSONBOARD',
```

Comment:
The backend doesn't really, since no account that exists in the DB should be being set to 'NEEDSONBOARD' from another value. My plan for implementation is to have accounts added to SWB be initialized to 'NEEDSONBOARD', but aside from that we shouldn't have an onboarded account that is suddenly becoming not onboarded.

If the stackname is missing, then the backend reads it as 'NEEDSONBOARD' if that's what the permissionsStatus is already set to, and 'NOSTACKNAME' otherwise.